### ISSUE TITLE
Feature - win_service plugin - only send state changes

### ISSUE THREAD

**Issue by cruscio** [influxdata/telegraf#6605]
### Proposal:

Add a configuration setting to win_service that will only output measures when the value of the `state` field for a service in the current collection differs from the `state` of the previous collection.

Or more generically, maybe a processor that drops a collection when the value of an enum field hasn't changed since the last collection.   

### Current behavior:

Each collection interval, win_service sends all services (or all services configured in `service_names`) to the configured output(s).
 
### Desired behavior:

- On the first collection interval, all measures are passed. 
- On subsequent intervals, only measures with a `state` different from the previous collection are output. 
- Maybe also have a setting to pass _all_ measures every _nth_ interval

### Use case:

My system has approx. 3k agents, 16 measures (almost all perfmon; including some large volume ones like process\\*), collected on a 1m interval, with 1.4MM series. We see approx 120k series written to Influx per minute. 

When we enabled win_service input on a 5m collection interval, we see points per minute jump to 250k-350k every 5 minutes. Volume more than doubles and all that extra data is mostly _"yup, everything's still the same"_. It caused too much IO for our cluster to handle and we started seeing timeouts in writes to Influx.

We'd like to be aware of service change events on a much smaller time-frame. Having a filter/processor that only passed changes would significantly reduce network and IO bandwidth.

**Comment by glinton** [comment]
> Maybe also have a setting to pass all measures every nth interval

I believe you can get the same results with the `flush_interval`

The other behavior I want to say there was talk of such a processor, possibly called "sample". I can't find any reference to it though.

> significantly reduce network and IO bandwidth

It may not work for you, but there is a [merge aggregator](https://github.com/influxdata/telegraf/tree/master/plugins/aggregators/merge) you could look at for now.

**Comment by cruscio** [comment]
> @glinton: I believe you can get the same results with the flush_interval

That would change `when` measurements are output, not `how many` measurements are output. I may not have been clear in my intent behind  _pass all measures every nth interval_.

Let's say you have a server with 200 services, a collection interval of 1m, an _nth_ interval on `win_service` of 10 minutes (or collections instead of time interval; whichever is easier to implement), and in that time service `foo` changes from running to stopped at minute 5. 

The desired outcome would be:

- Output at minute 1 consists of 200 measurements - one for each service
- No output at minutes 2, 3 & 4
- Output at minute 5 consists of a single measurement for service `foo`
- No output at minutes 6, 7, 8 & 9
- Output at minute 10 consists of 200 measurements - one for each service

Using `flush_interval=10m` would result in output of 2000 measurements at minute 10. The desired outcome would be only 401 measurements in total (200 measurements every 10 minutes, plus measurements for any changed service every minute). And as that scales, say for a collection interval of 1m with an nth interval of 60m - the difference becomes more significant: 12000 vs 401 measurements.

I'll take a look at `merge aggregator`, but I don't think it'll help the way I'm looking for here.

**Comment by danielnelson** [comment]
I agree that this should be implemented as a processor/aggregator, depending on how advanced we want to get.  A simple polling -> events conversion could be done as a processor, while some of the more advanced algorithms might make more sense as an aggregator since it has a periodic push trigger.

related #3388.
